pkg/bgpv1/gobgp/server.go

```diff
@@ -165,32 +164,44 @@ func (sc *ServerWithConfig) RemoveNeighbor(ctx context.Context, n *v2alpha1api.C
 // `ip` can be an ipv4 or ipv6 and this method will handle the differences
 // between MP BGP and BGP.
 //
-// `cstate` provides the IPv6 and IPv4 next host addresses which `must` be
-// Cilium's externally reachable IPv6 or IPv4 address.
-//
 // It is an error to advertise an IPv6 path when no IPv6 address is configured
 // on this Cilium node, selfsame for IPv4.
 //
+// Nexthop of the path will always set to "0.0.0.0" in IPv4 and "::" in IPv6, so
+// that GoBGP selects appropriate actual nexthop address and advertise it.
+//
 // An Advertisement is returned which may be passed to WithdrawPath to remove
 // this Advertisement.
-func (sc *ServerWithConfig) AdvertisePath(ctx context.Context, ip *net.IPNet, cstate *agent.ControlPlaneState) (Advertisement, error) {
+func (sc *ServerWithConfig) AdvertisePath(ctx context.Context, ip *net.IPNet) (Advertisement, error) {
 	var err error
 	var path *gobgp.Path
 	origin, _ := apb.New(&gobgp.OriginAttribute{
 		Origin: 0,
 	})
 	switch {
 	case ip.IP.To4() != nil:
-		if cstate.IPv4.IsUnspecified() {
-			return Advertisement{}, fmt.Errorf("cannot advertise IPv4 path %v, control plane has no configured IPv4 address for next hop", ip.String())
-		}
 		prefixLen, _ := ip.Mask.Size()
 		nlri, _ := apb.New(&gobgp.IPAddressPrefix{
 			PrefixLen: uint32(prefixLen),
 			Prefix:    ip.IP.String(),
 		})
+		// Currently, we only support advertising locally originated paths (the paths generated in Cilium
+		// node itself, not the paths received from another BGP Peer or redistributed from another routing
+		// protocols. In this case, the nexhop address should be the address used for peering. That means
+		// nexthop address varies depends on which neighbors we are  peering.
```

Comment:
Having a but of trouble understanding:

```
// protocols. In this case, the nexhop address should be the address used for peering. That means nexthop address varies depends on which neighbors we are  peering. 
```

also extra space between 'are' and 'peering'.

Reply:
I fixed the comment. Now is it ok with you?

Reply:
yup